src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1635,49 +1644,107 @@ private void VisitArguments(ImmutableArray<BoundExpression> arguments, Immutable
         private void VisitArgumentsWarn(
             ImmutableArray<BoundExpression> arguments,
             ImmutableArray<RefKind> refKindsOpt,
-            MethodSymbol method,
+            ImmutableArray<ParameterSymbol> parameters,
             ImmutableArray<int> argsToParamsOpt,
             bool expanded,
             ImmutableArray<Result> results)
         {
             for (int i = 0; i < arguments.Length; i++)
             {
-                RefKind refKind = GetRefKind(refKindsOpt, i);
-                var argument = arguments[i];
-                var parameter = GetCorrespondingParameter(i, method, argsToParamsOpt, ref expanded);
-                var result = results[i];
-                if (refKind != RefKind.None)
+                bool expandedParameter = expanded;
+                var parameter = GetCorrespondingParameter(i, parameters, argsToParamsOpt, ref expandedParameter);
+                if (parameter is null)
                 {
-                    var parameterType = parameter?.Type;
-                    ReportAssignmentWarnings(argument, result.Type, parameterType, useLegacyWarnings: false);
-                    TrackNullableStateForAssignment(argument, result.Type, result.Slot, parameterType, -1);
-                }
-                if (refKind != RefKind.Out && (object)parameter != null)
-                {
-                    WarnOnNullReferenceArgument(argument, result.Type, parameter, expanded);
+                    continue;
                 }
+                VisitArgumentWarn(
+                    arguments[i],
+                    GetRefKind(refKindsOpt, i),
+                    parameter,
+                    expandedParameter,
+                    results[i]);
             }
         }
 
-        private static ParameterSymbol GetCorrespondingParameter(int argumentOrdinal, MethodSymbol method, ImmutableArray<int> argsToParamsOpt, ref bool expanded)
+        private void VisitArgumentWarn(
+            BoundExpression argument,
+            RefKind refKind,
+            ParameterSymbol parameter,
+            bool expanded,
+            Result result)
         {
-            if ((object)method == null)
+            var parameterType = GetParameterType(parameter, expanded);
+            var conversion = GenerateConversion(_conversions, argument, result.Type?.TypeSymbol, parameterType.TypeSymbol);
```

Comment:
>GenerateConversion [](start = 29, length = 18)

Is it fine to do this even for cases when we did not remove conversions?

Reply:
Yes, and it's necessary, since a value of type `C<object>` will be passed to a parameter of type `C<object?>` with no implicit conversion, but we'll want to warn here.

---
In reply to: [181200827](https://github.com/dotnet/roslyn/pull/26107#discussion_r181200827) [](ancestors = 181200827)